fhir_extractor/src/main.py

```diff
@@ -2,89 +2,58 @@
 
 import os
 import json
-from flask import Flask
-from flask import request
-from flask_restful import Resource, Api, reqparse
-from flask_sqlalchemy import SQLAlchemy
+from confluent_kafka import KafkaException, KafkaError
+from sqlalchemy import create_engine
 
 from fhir_extractor.src.extract import Extractor
 from fhir_extractor.src.analyze import Analyzer
 
 from fhir_extractor.src.analyze.mapping import get_mapping
 from fhir_extractor.src.analyze.graphql import get_resource_from_id
 from fhir_extractor.src.config.logger import create_logger
-from fhir_extractor.src.config.database_config import DatabaseConfig
+from fhir_extractor.src.config.database_config import DatabaseConfig, get_db_url
 from fhir_extractor.src.errors import OperationOutcome
 from fhir_extractor.src.helper import get_topic_name
 from fhir_extractor.src.producer_class import ExtractorProducer
+from fhir_extractor.src.consumer_class import ExtractorConsumer
 
 logger = create_logger("fhir_extractor")
 
-# Create flask app object
-app = Flask(__name__)
-api = Api(app)
-
-app.config.from_object(DatabaseConfig)
-db = SQLAlchemy(app)
-
+db_engine = create_engine(get_db_url())
 analyzer = Analyzer()
-extractor = Extractor(engine=db.engine)
+extractor = Extractor(engine=db_engine)
 producer = ExtractorProducer(broker=os.getenv("KAFKA_BOOTSTRAP_SERVERS"))
 
 
-@app.route("/batch", methods=["POST"])
-def extractor_sql_batch():
-    """
-    Extract all records for the specified resource
+def process_event(msg):
     """
-    body = request.get_json()
-    resource_ids = body.get("resourceIds", None)
-    logger.debug("Params obtained")
-    try:
-        # Get the resources we want to process from the pyrog mapping for a given source
-        logger.debug("Getting Mapping")
-        resources = get_mapping(resource_ids=resource_ids)
-
-        for resource_mapping in resources:
-            logger.debug("Analysing Resource")
-            analysis = analyzer.analyze(resource_mapping)
 
-            run_resource(resource_mapping, analysis)
-
-        return "Success", 200
+    :return:
+    """
+    msg_value = json.loads(msg.value())
+    resource_id = msg_value.get('resource_id', None)
+    primary_key_values = msg_value.get('primaryKeyValues', None)
```

Comment:
When I call the `run_sample/` endpoint, it runs the whole batch instead of just the primary key value specified. 
This is because of `primaryKeyValues` instead of `primary_key_values` in the event produced by the API

Reply:
By the way, I haven't been very good with this casing convention I think so if you have any idea of how to improve that, feel free!

Reply:
What about using snake_case everywhere?

Reply:
Fine for me !

Reply:
Then we'll need to use snake case in the request body from Pyrog?
I'm ok if it's not anti-conventional 😄